generate_aliases/pkg/generate/generate.go

```diff
@@ -0,0 +1,122 @@
+package generate
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/Dbz/kube-aliases/pkg/types"
+)
+
+func Generate(filePath, targetPath string) error {
+	file, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return fmt.Errorf("error reading file %s with error %s",
+			filePath, err)
+	}
+
+	var aliases types.Aliases
+	err = yaml.Unmarshal(file, &aliases)
+	if err != nil {
+		return fmt.Errorf("error unmarshaling file: %v", err)
+	}
+
+	aliasFile, err := os.Create(targetPath)
+	if err != nil {
+		return fmt.Errorf("Failed to create alias file: %v", err)
+	}
+	defer aliasFile.Close()
+
+	var aliasBuilder strings.Builder
+	aliasNames := map[string]string{}
+	for r := range aliases.Resources {
+
+		// Generate Commands
+		for c := range aliases.CMDs {
+
+			err := GenerateAlias(&aliasBuilder,
+				aliasNames,
+				&types.AliasCMD{
+					PrefixShort:   aliases.CMDs[c].Prefix.Short,
+					ResourceShort: aliases.Resources[r].Short,
+					Short:         aliases.CMDs[c].Short,
+					SuffixShort:   aliases.CMDs[c].Suffix.Short,
+					Prefix:        aliases.CMDs[c].Prefix.CMD,
+					CMD:           aliases.CMDs[c].CMD,
+					Resource:      r,
+					Suffix:        aliases.CMDs[c].Suffix.CMD,
+				})
+			if err != nil {
+				return err
+			}
+
+		}
+
+		for _, v := range aliases.Resources[r].AdditonalCMDs {
+
+			err := GenerateAlias(&aliasBuilder,
+				aliasNames,
+				&types.AliasCMD{
+					PrefixShort:   v.Prefix.Short,
+					ResourceShort: aliases.Resources[r].Short,
+					Short:         v.Short,
+					SuffixShort:   v.Suffix.Short,
+					Prefix:        v.Prefix.CMD,
+					CMD:           v.CMD,
+					Resource:      r,
+					Suffix:        v.Suffix.CMD,
+				})
+
+			if err != nil {
+				return err
+			}
+		}
+
+	}
+
+	_, err = io.WriteString(aliasFile, aliasBuilder.String())
+	if err != nil {
+		return fmt.Errorf("Warning: could not write alias: %v\n", err)
+	}
+
+	return nil
+
+}
+
+// GenerateAlias -- TODO document
+func GenerateAlias(w io.Writer, aliasNames map[string]string, alias *types.AliasCMD) error {
+
+	if alias.Prefix != "" {
+		alias.Prefix = alias.Prefix + " "
+	}
+	if alias.Suffix != "" {
+		alias.Suffix = " " + alias.Suffix
```

Comment:
Same suggestion here. Def no pressure to accept suggestion
```suggestion
		alias.Suffix = " " + strings.Trim(alias.Suffix, " ")
```

Reply:
Yep